### PR TITLE
fix(compiler): Fix compilation of functions annotated with alias

### PR DESCRIPTION
**Pull request by ospencer**
Added some regression tests!

### CODE REVIEW

File: compiler/src/typed/ctype.re [hunk 1]
```diff
@@ -3246,17 +3246,8 @@ let nondep_extension_constructor = (env, id, ext) =>
...
-              Type_utils.wasm_repr_of_allocation_type(
-                Type_utils.get_allocation_type(env, arg),
-              ),
-            args,
-          ),
-          [WasmI32],
-          Indirect,
-        )
+        // All native Grain function args are Managed (i32) types.
+        ReprFunction(List.map(_ => WasmI32, args), [WasmI32], Indirect)
```
Comment (ospencer) [inline compiler/src/typed/ctype.re:3250]: Had to do this to get rid of the circular dep—it's correct anyway, but I added a comment just in case.

**Review by phated — APPROVED** [review]
Excellent fix!

**Review by marcusroberts — APPROVED** [review]
Super quick fix!

**Review by phated — APPROVED** [review]
Reapproved!

**Pull request merged**